machineconfiguration/v1alpha1/types_osimagestream.go

```diff
@@ -0,0 +1,124 @@
+	// +required
+	Spec *OSImageStreamSpec `json:"spec,omitempty"`
+
+	// status describes the last observed state of this OSImageStream.
+	// Populated by the MachineConfigOperator after reading release metadata.
+	// When not present, the controller has not yet reconciled this resource.
+	// +optional
+	Status *OSImageStreamStatus `json:"status,omitempty"`
+}
+
+// OSImageStreamStatus describes the current state of a OSImageStream.
+// +kubebuilder:validation:XValidation:rule="!has(self.availableStreams) || size(self.availableStreams) == 0 || (has(self.defaultStream) && size(self.defaultStream) != 0)",message="defaultStream must be set when availableStreams is not empty"
+// +kubebuilder:validation:XValidation:rule="!has(self.defaultStream) || self.defaultStream in self.availableStreams.map(s, s.name)",message="defaultStream must reference a stream name from availableStreams"
+type OSImageStreamStatus struct {
+	// availableStreams is a list of the available OS Image Streams
+	// available and their associated URLs for both OS and Extensions
+	// images.
+	//
+	// A maximum of 100 streams may be specified.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
+	// +listType=map
+	// +listMapKey=name
+	AvailableStreams []OSImageStreamURLSet `json:"availableStreams,omitempty"`
+
+	// defaultStream is the name of the stream that should be used as the default
+	// when no specific stream is requested by a MachineConfigPool.
+	// Must reference the name of one of the streams in availableStreams.
+	// Must be set when availableStreams is not empty.
+	// When not set and availableStreams is empty, controllers should use the default one stated in the release image.
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=70
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^[\\w\\.\\-]+$')`,message="The name must consist only of alphanumeric characters, hyphens ('-') and dots ('.')."
+	DefaultStream string `json:"defaultStream,omitempty"`
+}
+
+// OSImageStreamSpec defines the desired state of a OSImageStream.
+type OSImageStreamSpec struct {
+}
+
+type OSImageStreamURLSet struct {
+	// name is the identifier of the stream.
+	//
+	// Must not be empty and must not exceed 70 characters in length.
+	// Must only contain alphanumeric characters, hyphens ('-'), or dots ('.').
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=70
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^[\\w\\.\\-]+$')`,message="The name must consist only of alphanumeric characters, hyphens ('-') and dots ('.')."
+	Name string `json:"name,omitempty"`
+
+	// osImageURL is an OS Image referenced by digest.
+	//
+	// The format of the URL ref is:
+	// host[:port][/namespace]/name@sha256:<digest>
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=447
```

Comment:
We have no docs on that max length. Also, how is that calculated? 

The same applies to `OSExtensionsImageURL`.